components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1690 @@
+
+        return new Object[][]{
+                {"testIdP1", 1, SAMPLE_TENANT_ID, 2},
+                {"testIdP2", 2, SAMPLE_TENANT_ID, 0},
+        };
+    }
+
+    @Test(dataProvider = "getPermissionsAndRoleConfigurationData")
+    public void testGetPermissionsAndRoleConfiguration(String idpName, int idpId, int tenantId, int resultCount)
+            throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            PermissionsAndRoleConfig pac = idPManagementDAO.getPermissionsAndRoleConfiguration(connection, idpName,
+                    idpId, tenantId);
+            assertEquals(pac.getIdpRoles().length, resultCount);
+        }
+    }
+
+    @DataProvider
+    public Object[][] getProvisioningConnectorConfigsData() {
+
+        return new Object[][]{
+                {"testIdP1", 1, SAMPLE_TENANT_ID, 2},
+                {"testIdP2", 2, SAMPLE_TENANT_ID, 0},
+        };
+    }
+
+    @Test(dataProvider = "getProvisioningConnectorConfigsData")
+    public void testGetProvisioningConnectorConfigs(String idpName, int idpId, int tenantId, int resultCount)
+            throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            ProvisioningConnectorConfig[] pccResult = idPManagementDAO.getProvisioningConnectorConfigs(connection,
+                    idpName, idpId, tenantId);
+            assertEquals(pccResult.length, resultCount);
+        }
+    }
+
+    @DataProvider
+    public Object[][] getIdPByNameData() {
+
+        return new Object[][]{
+                {"testIdP1", SAMPLE_TENANT_ID, true},
+                {"testIdP3", SAMPLE_TENANT_ID2, true},
+                {"notExist", SAMPLE_TENANT_ID, false},
+        };
+    }
+
+    @Test(dataProvider = "getIdPByNameData")
+    public void testGetIdPByName(String idpName, int tenantId, boolean isExist) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+
+            //getIdPByName
```

Comment:
Do we need this comment?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/commit/6b030460eec87be9f79276256c9a3617fe357398